rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -769,6 +771,88 @@ void main_window::HandlePackageInstallation(QStringList file_paths)
 	}
 }
 
+void main_window::ExtractMSELF()
+{
+	const QString path_last_mself = m_gui_settings->GetValue(gui::fd_extract_mself).toString();
+	QString file_path = QFileDialog::getOpenFileName(this, tr("Select MSELF To extract"), path_last_mself, tr("All pup files (*.mself);;All files (*.*)"));
+
+	QString dir = file_path.isEmpty() ? QString{} : QFileDialog::getExistingDirectory(this, tr("Extraction Directory"), QString{}, QFileDialog::ShowDirsOnly | QFileDialog::DontResolveSymlinks);
+
+	if (!dir.isEmpty())
+	{
+		HandleMSELFextraction(file_path, dir);
+	}
+}
+
+void main_window::HandleMSELFextraction(QString file_path, QString dir_path)
```

Comment:
why is this part of the main window?
There is no reason to implement this in a gui dialog

Reply:
Where'd you put it?

Reply:
mself obviously